src/util/mod.rs

```diff
@@ -248,14 +248,31 @@ pub fn clean_dead_namespaces() -> anyhow::Result<()> {
 }
 
 pub fn elevate_privileges() -> anyhow::Result<()> {
+    use signal_hook::{cleanup, flag, SIGINT};
+    use std::sync::atomic::{AtomicBool, Ordering};
+    use std::sync::Arc;
+
     // Check if already running as root
     if nix::unistd::getuid().as_raw() != 0 {
         info!("Calling sudo for elevated privileges, current user will be used as default user");
         let args: Vec<String> = std::env::args().collect();
 
+        let terminated = Arc::new(AtomicBool::new(false));
+        flag::register(SIGINT, Arc::clone(&terminated))?;
+
```

Comment:
Handling SIGINT here is necessary since this is a separate process from the one that actually executes.
Otherwise, the elevated process becomes orphaned.